### PR TITLE
MAINT: Update code from ADO - 0.6.0.dev8

### PR DESCRIPTION
**Pull request by rchopade7**
MAINT: Update code from ADO - 0.6.0.dev8

### CODE REVIEW

**Review by waltersma — APPROVED** [review]

File: src/ansys/meshing/prime/autogen/connect.py [hunk 1]
```diff
@@ -126,13 +126,13 @@ def join_face_zonelets(self, part_id : int, face_zonelet_ids : Iterable[int], wi
         return ConnectResults(model = self._model, json_data = result)
 
     def subtract_volumes(self, part_id : int, target_volumes : Iterable[int], cutter_volumes : Iterable[int], params : SubtractVolumesParams) -> SubtractVolumesResults:
-        """ Subtract cutter volumes from target volumes. Volumes should be computed prior to calling this function.
+        """ Subtract cutter volumes from target volumes. Volumes should be computed prior to calling this function. If multiple parts are being merged to form a single part, then volumes should be computed for each part prior to merging. Use compute_closed_volumes to do so.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connect.py:129]:
```suggestion
        """ Subtract cutter volumes from target volumes. Volumes should be computed prior to calling this function. If multiple parts are being merged to form a single part, then volumes should be computed for each part prior to merging. Use the ``compute_closed_volumes()`` method to do so.
```

File: src/ansys/meshing/prime/autogen/connect.py [hunk 2]
```diff
@@ -126,13 +126,13 @@ def join_face_zonelets(self, part_id : int, face_zonelet_ids : Iterable[int], wi
...
 
 
         Parameters
         ----------
         part_id : int
-            Id of a part.
+            Id of part containing target and cutter volumes for subtract operation.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connect.py:135]:
```suggestion
            ID of the part containing the target and cutter volumes for the subtract operation.
```
While we omit the article at the start of the brief description, we use them inside the description if it improves readability.

File: src/ansys/meshing/prime/autogen/connect.py [hunk 3]
```diff
@@ -126,13 +126,13 @@ def join_face_zonelets(self, part_id : int, face_zonelet_ids : Iterable[int], wi
...
         target_volumes : Iterable[int]
             Ids of target volumes.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connect.py:137]:
```suggestion
            IDs of target volumes.
```

File: src/ansys/meshing/prime/autogen/connectstructs.py [hunk 4]
```diff
@@ -856,7 +856,7 @@ def __init__(
         ignore_face_zonelets: Iterable[int], optional
             Face zonelet ids that subtract volumes should not remove (for example, periodic or fluid cap zonelets). If ignore face zonelets are provided, then the target volumes after subtract operation need to be recomputed.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connectstructs.py:857]:
```suggestion
            Face zonelet IDs that subtract volumes should not remove (for example, periodic or fluid cap zonelets). If ignore face zonelets are provided, then the target volumes after the subtract operation must be recomputed.
```

File: src/ansys/meshing/prime/autogen/connectstructs.py [hunk 5]
```diff
@@ -856,7 +856,7 @@ def __init__(
...
         check_cutters: bool, optional
-            Option to analyze overlapping or intersecting cutter volumes.
+            Option to manage intersecting cutter volumes. When keep_cutters is False an error message is provided if multiple cutters intersect. Overlapping cutter volumes are not supported.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connectstructs.py:859]:
```suggestion
            Whether to check intersecting cutter volumes. The default is ``None``. When ``keep_cutters=False``, an error message is raised if multiple cutters intersect. Overlapping cutter volumes are not supported.
```

File: src/ansys/meshing/prime/autogen/connectstructs.py [hunk 6]
```diff
@@ -856,7 +856,7 @@ def __init__(
...
         keep_cutters: bool, optional
             Option to retain cutter volumes.
             This is a beta parameter. The behavior and name may change in the future.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connectstructs.py:862]:
```suggestion
            Whether to retain cutter volumes.
            This is a beta parameter. The behavior and name may change in the future.
```

File: src/ansys/meshing/prime/autogen/connectstructs.py [hunk 7]
```diff
@@ -908,7 +908,7 @@ def set_default(
         ignore_face_zonelets: Iterable[int], optional
             Face zonelet ids that subtract volumes should not remove (for example, periodic or fluid cap zonelets). If ignore face zonelets are provided, then the target volumes after subtract operation need to be recomputed.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connectstructs.py:909]:
```suggestion
            Face zonelet IDs that subtract volumes should not remove (for example, periodic or fluid cap zonelets). If values are provided for the ``ignore_face_zonelets`` parameter, then the target volumes after the subtract operation must be recomputed.
```
In descriptive text, "ids" shoud always be "IDs".

File: src/ansys/meshing/prime/autogen/connectstructs.py [hunk 8]
```diff
@@ -908,7 +908,7 @@ def set_default(
...
         check_cutters: bool, optional
-            Option to analyze overlapping or intersecting cutter volumes.
+            Option to manage intersecting cutter volumes. When keep_cutters is False an error message is provided if multiple cutters intersect. Overlapping cutter volumes are not supported.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connectstructs.py:911]:
```suggestion
            Whether to retain intersecting cutter volumes. When ``keep_cutters= False``, an error message is raised if multiple cutters intersect. Overlapping cutter volumes are not supported.
```

File: src/ansys/meshing/prime/autogen/connectstructs.py [hunk 9]
```diff
@@ -908,7 +908,7 @@ def set_default(
...
         keep_cutters: bool, optional
             Option to retain cutter volumes.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connectstructs.py:913]:
```suggestion
            Whether to retain cutter volumes.
```

File: src/ansys/meshing/prime/autogen/connectstructs.py [hunk 10]
```diff
@@ -955,7 +955,7 @@ def ignore_face_zonelets(self, value: Iterable[int]):
 
     @property
     def check_cutters(self) -> bool:
-        """Option to analyze overlapping or intersecting cutter volumes.
+        """Option to manage intersecting cutter volumes. When keep_cutters is False an error message is provided if multiple cutters intersect. Overlapping cutter volumes are not supported.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/connectstructs.py:958]:
```suggestion
        """Flag indicating if intersecting cutter volumes are checked. When ``keep_cutters=False``, an error message is raised if multiple cutters intersect. Overlapping cutter volumes are not supported.
```

File: src/ansys/meshing/prime/autogen/primeconfig.py [hunk 11]
```diff
@@ -651,6 +651,15 @@ class ErrorCode(enum.IntEnum):
...
+    This is a beta parameter. The behavior and name may change in the future."""
+    ZERONODESREADFROMCDBFILE = 16501
+    """No nodes read from CDB file.
+    This is a beta parameter. The behavior and name may change in the future."""
+    ZEROELEMENTSFORCDBEXPORT = 16600
+    """No elements found for cdb export.
+    This is a beta parameter. The behavior and name may change in the future."""
 
 class WarningCode(enum.IntEnum):
     """Warning codes associated with the PyPrimeMesh operation.
```
Comment (PipKat) [inline src/ansys/meshing/prime/autogen/primeconfig.py:665]:
```suggestion
    """Provides warning codes associated with the PyPrimeMesh operation.
```

File: src/ansys/meshing/prime/internals/error_handling.py [hunk 12]
```diff
@@ -47,7 +47,7 @@
     ErrorCode.PERIODICSURFACESNOTSUPPORTEDFORPRISMS: "Periodic surfaces selected for prism generation, not supported.",
     ErrorCode.ALREADYVOLUMEMESHED: "Already volume meshed.",
     ErrorCode.INCREMENTALVOLUMEMESHINGNOTSUPPORTED: "Incremental volume meshing is not supported.",
-    ErrorCode.VOLUMESNOTUPTODATE: "Volumes are not up to date. Update volumes and try again.",
+    ErrorCode.VOLUMESNOTUPTODATE: "Volumes are not up to date. Compute topovolumes for topology part or compute closed volumes for non topology parts and try again.",
```
Comment (PipKat) [inline src/ansys/meshing/prime/internals/error_handling.py:50]:
```suggestion
    ErrorCode.VOLUMESNOTUPTODATE: "Volumes are not up to date. Compute topovolumes for topology part or compute closed volumes for non-topology parts and try again.",
```

File: src/ansys/meshing/prime/internals/error_handling.py [hunk 13]
```diff
@@ -47,7 +47,7 @@
...
     ErrorCode.QUADRATICMESHSUPPORTEDONLYFORTETS: "Quadratic meshing is supported only for tetrahedrons.",
     ErrorCode.NOACTIVESFFOUND: "Active size fields are not available.",
     ErrorCode.AITOVERLAPALONGMULTIFOUND: "Overlapping faces along multi-connection found.",
```
Comment (PipKat) [inline src/ansys/meshing/prime/internals/error_handling.py:53]:
```suggestion
    ErrorCode.AITOVERLAPALONGMULTIFOUND: "Found overlapping faces along multi-connection.",
```

File: src/ansys/meshing/prime/internals/error_handling.py [hunk 14]
```diff
@@ -308,6 +308,9 @@
     ErrorCode.AUTOQUADMESHER_INVALIDMINMAXSIZES: "Minimum size is more than maximum size.",
     ErrorCode.AUTOQUADMESHER_NEGATIVEINPUTPARAMETER: "Input parameters contain one or more negative values.",
     ErrorCode.FACEZONELETSHAVECELLSCONNECTED: "Face zonelets have cells connected.",
+    ErrorCode.ZEROELEMENTSREADFROMCDBFILE: "No mesh elements found. Check the input CDB file.",
```
Comment (PipKat) [inline src/ansys/meshing/prime/internals/error_handling.py:311]:
```suggestion
    ErrorCode.ZEROELEMENTSREADFROMCDBFILE: "No mesh elements were found. Check the input CDB file.",
```

File: src/ansys/meshing/prime/internals/error_handling.py [hunk 15]
```diff
@@ -308,6 +308,9 @@
...
+    ErrorCode.ZERONODESREADFROMCDBFILE: "No nodes found. Check the input CDB file.",
```
Comment (PipKat) [inline src/ansys/meshing/prime/internals/error_handling.py:312]:
```suggestion
    ErrorCode.ZERONODESREADFROMCDBFILE: "No nodes were found. Check the input CDB file.",
```

File: src/ansys/meshing/prime/internals/error_handling.py [hunk 16]
```diff
@@ -308,6 +308,9 @@
...
+    ErrorCode.ZEROELEMENTSFORCDBEXPORT: "No mesh elements found for CDB export. Check if the model is meshed, or set write_by_zones in ExportMapdlCdbParams to false if zones are not defined.",
```
Comment (PipKat) [inline src/ansys/meshing/prime/internals/error_handling.py:313]:
```suggestion
    ErrorCode.ZEROELEMENTSFORCDBEXPORT: "No mesh elements were found for CDB export. Check if the model is meshed, or set  'write_by_zones' in 'ExportMapdlCdbParams' to 'False' if zones are not defined.",
```

**Review by prajaktmahajan — APPROVED** [review]

**Review by waltersma — APPROVED** [review]

File: src/ansys/meshing/prime/autogen/connect.py (hunk 1 above)
Comment (rchopade7) [reply src/ansys/meshing/prime/autogen/connect.py:129]: @PipKat I will pick this in next github PR

**Comment by rchopade7** [comment]
@PipKat  I will pick all doc review comments in next github PR

**Pull request merged**